src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,728 @@
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"
+using System;
+
+class C
+{
+    void M()
+    {
+        local("""");
+        void local(string s)
+        {
+            [|if (s " + @operator + @" null)
+            {
+                throw new ArgumentNullException(nameof(s));
+            }|]
+        }
+    }
+}",
+                FixedCode = @"
+using System;
+
+class C
+{
+    void M()
+    {
+        local("""");
+        void local(string s!!)
+        {
+        }
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestEqualitySwapped()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"
+using System;
+
+class C
+{
+    void M(string s)
+    {
+        [|if (null == (object)s)
+            throw new ArgumentNullException(nameof(s));|]
+    }
+}",
+                FixedCode = @"
+using System;
+
+class C
+{
+    void M(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotEquality()
+        {
+            var testCode = @"
+using System;
+
+class C
+{
+    void M(string s)
+    {
+        if ((object)s != null)
+            throw new ArgumentNullException(nameof(s));
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNullCoalescingThrow()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"
+using System;
+
+class C
+{
+    private readonly string s;
+    public C(string s)
+    {
+        [|this.s = s ?? throw new ArgumentNullException(nameof(s));|]
+    }
+}",
+                FixedCode = @"
+using System;
+
+class C
+{
+    private readonly string s;
+    public C(string s!!)
+    {
+        this.s = s;
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestReferenceEqualsCheck()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"
+using System;
+
+class C
+{
+    private readonly string s;
+    void M(string s)
+    {
+        [|if (object.ReferenceEquals(s, null))
+        {
+            throw new ArgumentNullException(nameof(s));
+        }|]
```

Comment:
The diagnostic span is too large that I think it could be noisy. I think reporting on the ParameterSyntax is less nosiy, and can actually much simplify the codefix implementation. @CyrusNajmabadi What do you think?

Reply:
Do you mean reporting on the parameter declaration or the usage?

Reply:
I was thinking of the declaration (ie, swap the diagnostic location with the `additionalLocations`). Implementation-wise, there shouldn't be much of changes. The benefit is mainly a smaller span which I *think* is better user experience. But I'm not sure if there are any guidelines the IDE team uses around diagnostic spans.

Reply:
We discussed it offline. General idea here is that while this is large, it's actually fixed size (not unbounded).

Reply:
I'm happy with it as written because it covers a bounded part of the code that will be eliminated by the fix. Going by that, though, it might be reasonable to change the `field = param ?? throw ...` scenario to squiggle just the `throw ...` part.